zerver/lib/test_classes.py

```diff
@@ -167,6 +184,35 @@ def run(self, result: Optional[TestResult]=None) -> Optional[TestResult]:  # noc
     DEFAULT_SUBDOMAIN = "zulip"
     TOKENIZED_NOREPLY_REGEX = settings.TOKENIZED_NOREPLY_EMAIL_ADDRESS.format(token="[a-z0-9_]{24}")
 
+    @contextmanager
+    def assertNoLogs(self, test_case,logger_name, level, no_logs, exc_type, exc_value,tb):
```

Comment:
`level` and `logger_name` are the only two parameter that should be mandatory. You can keep rest other parameters as optional parameters. It seems you have not utilized most of these parameters in the function, so it makes sense to try removing them or making them optional.

Reply:
Please have a look at new solution posted.